runtime/handler.go

```diff
@@ -100,13 +125,23 @@ func handleForwardResponseOptions(ctx context.Context, w http.ResponseWriter, re
 	return nil
 }
 
-func handleForwardResponseStreamError(w http.ResponseWriter, err error) {
+func handleForwardResponseStreamError(ctx context.Context, w http.ResponseWriter, err error) {
+	md, ok := ServerMetadataFromContext(ctx)
+	if !ok {
+		glog.Errorf("Failed to extract ServerMetadata from context")
+		return
+	}
+
 	grpcCode := grpc.Code(err)
 	httpCode := HTTPStatusFromCode(grpcCode)
-	resp := responseStreamChunk{Error: &responseStreamError{GrpcCode: int(grpcCode),
-		HTTPCode:   httpCode,
-		Message:    err.Error(),
-		HTTPStatus: http.StatusText(httpCode)}}
+	resp := responseStreamChunk{
+		Error: &ResponseStreamError{
+			GrpcCode:   int(grpcCode),
+			HTTPCode:   httpCode,
+			Message:    err.Error(),
+			HTTPStatus: http.StatusText(httpCode),
+			Trailer:    md.TrailerMD,
```

Comment:
IIUC, the trailer does not belong to this specific chunk but it indicates an application-level error of entire the stream.
So I don't think it is a good idea to put the trailer here.